server/src/controllers/client/getAllClients.js

```diff
@@ -1,9 +1,104 @@
 const logger = require("pino")();
 const Client = require("../../models/client.model");
+const sequelize = require("sequelize");
+const Op = sequelize.Op;
 
 const getAllClientsRequestHandler = async (req, res) => {
   try {
-    const clients = await Client.findAll();
+    const {
+      name,
+      phone_number,
+      date_updated_from,
+      date_updated_to,
+      date_registered_from,
+      date_registered_to,
+      owner,
+      active,
+      r_and_i,
+      closed,
+    } = req.query;
+
+    let query = {};
+    let query_options = {};
+
+    // build the query object
+    if (name) {
+      query = {
+        ...query,
+        [Op.and]: sequelize.literal(`LOWER(name) LIKE :name`),
+      };
+      query_options = {
+        replacements: { name: `%${name.toLowerCase()}%` },
+      };
+    }
+
+    if (phone_number) {
+      query.phone_number = { [Op.like]: `%${phone_number}%` };
+    }
+
+    if (date_updated_from) {
+      query.date_updated = {
+        [Op.gte]: new Date(date_updated_from),
+      };
+    }
+
+    if (date_updated_to) {
+      const date = new Date(date_updated_to);
+      date.setUTCHours(23, 59, 59, 999);
+      query.date_updated = {
+        ...query.date_updated,
+        [Op.lte]: date,
+      };
+    }
+
+    if (date_registered_from) {
+      query.date_added = {
+        [Op.gte]: new Date(date_registered_from),
+      };
+    }
+
+    if (date_registered_to) {
+      const date = new Date(date_registered_to);
+      date.setHours(23, 59, 59, 999);
+      query.date_added = {
+        ...query.date_added,
+        [Op.lte]: date,
+      };
+    }
+
+    if (owner) {
+      query.owner = owner;
+    }
+    let status_options = [];
+    if (active === "true") {
+      status_options = ["active"];
+    }
+    if (r_and_i === "true") {
+      status_options = [...status_options, "r_and_i"];
+    }
+    if (closed === "true") {
+      status_options = [...status_options, "closed"];
+    }
+
+    query.status = {
+      [Op.or]: status_options,
+    };
+
+    query_options = {
+      ...query_options,
+      where: query,
+    };
+    const page = Number(req.query.page);
+    const pageSize = Number(req.query.pageSize);
+    if (page && pageSize) {
+      query_options = {
+        ...query_options,
+        limit: pageSize,
+        offset: page * pageSize,
+      };
+    }
+
+    const clients = await Client.findAndCountAll(query_options);
 
     return res.status(200).json({
```

Comment:
within the response body can you also include a list of all unique owners here is an example

```
const uniqueOwners = await JobLead.findAll({
      attributes: [[Sequelize.fn("DISTINCT", Sequelize.col("owner")), "owner"]],
      raw: true,
    });
    const uniqueOwnersList = Array.isArray(uniqueOwners)
      ? uniqueOwners.map((owner) => owner.owner)
      : [];
```

and then attach it to the reponse body like 

```
return res.status(200).json({
      status: "success",
      message: "All clients found successfully",
      data: clients,
      uniqueOwners: uniqueOwnersList,
    });
```